src/configure/configure.ts

```diff
@@ -443,16 +443,41 @@ class Orchestrator {
         //Post selecting the template update this.inputs.repositoryAnalysisApplicationSettings with corresponding languageSettings
         if (!!repoAnalysisResult
             && !!repoAnalysisResult.repositoryAnalysisApplicationSettingsList) {
-
+            
             //Get languageSettings (corresponding to language of selected settings) provided by RepoAnalysis
-            this.inputs.repositoryAnalysisApplicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.find(applicationSettings => {
-                return applicationSettings.language === this.inputs.pipelineConfiguration.template.language;
-            });
+            this.updateRepositoryAnalysisApplicationSettings(repoAnalysisResult);
         }
 
         telemetryHelper.setTelemetry(TelemetryKeys.ChosenTemplate, this.inputs.pipelineConfiguration.template.label);
     }
 
+    private async updateRepositoryAnalysisApplicationSettings(repoAnalysisResult: RepositoryAnalysisParameters): Promise<void>{
+        var applicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.filter(applicationSetting => {
```

Comment:
You should also handle the case where no applicationSetting matches and then trace an error

Reply:
Added a fail fast check